salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,326 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.wso2.carbon.inbound.endpoint.protocol.generic.GenericPollingConsumer;
+
+import org.cometd.bayeux.Channel;
+import org.cometd.bayeux.Message;
+import org.cometd.bayeux.client.ClientSessionChannel;
+import org.cometd.bayeux.client.ClientSessionChannel.MessageListener;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.ClientTransport;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.ContentExchange;
+import org.eclipse.jetty.client.HttpClient;
+
+import java.lang.Exception;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Salesforce streaming api Inbound.
+ */
+public class SalesforceStreamData extends GenericPollingConsumer {
+
+    private static final Log log = LogFactory.getLog(SalesforceStreamData.class);
+
+    // Mandatory parameters.
+    private static final boolean version = false;
+    private static final boolean useCookies = version;
+    protected static String loginEndpoint;
+
+    private String userName;
+    private String password;
+    private String salesforceObject;
+    private String channel;
+    private String packageVersion;
+    private String packageName;
+    private String streamingEndpointUri;
+    private String injectingSeq;
+
+    // optional parameters.
+    private static int connectionTimeout;
+    private static int readTimeout;
+    private static int waitTime;
+
+    private BayeuxClient client = null;
+    private HttpClient httpClient;
+
+    private boolean isPolled = false;
+
+    public SalesforceStreamData(Properties salesforceProperties, String name,
+                                SynapseEnvironment synapseEnvironment, long scanInterval,
+                                String injectingSeq, String onErrorSeq, boolean coordination,
+                                boolean sequential) {
+        super(salesforceProperties, name, synapseEnvironment, scanInterval,
+                injectingSeq, onErrorSeq, coordination, sequential);
+        loadMandatoryParameters(salesforceProperties);
+        loadOptionalParameters(salesforceProperties);
+        streamingEndpointUri = version ? "/" + packageName : "/" + packageName + "/" + packageVersion;
+        channel = version ? "/" + salesforceObject : "/topic/" + salesforceObject;
+        this.injectingSeq = injectingSeq;
+        httpClient = new HttpClient();
+    }
+
+    private void makeConnect() {
+        try {
+            client = makeClient();
+        } catch (Exception e) {
+            handleException(e.getMessage());
+        }
+        client.getChannel(Channel.META_HANDSHAKE).addListener(new ClientSessionChannel.MessageListener() {
```

Comment:
Here, May be changes to produce NullpointException since makeClient() can return null. Please do null check.